### PR TITLE
Make session an AttrDict

### PR DESCRIPTION
**Pull request by jesper-friis**
Ensure that the session is an AttrDict

### CODE REVIEW

File: otelib/pipe.py [hunk 1]
```diff
@@ -13,6 +15,6 @@ class Pipe:
     def __init__(self, strategy: "AbstractStrategy") -> None:
         self.input: "AbstractStrategy" = strategy
 
-    def get(self, session: "Optional[str]" = None) -> bytes:
+    def get(self, session: "Optional[AttrDict]" = None) -> bytes:
```
Comment (CasperWA) [inline otelib/pipe.py:18]: This is wrong, it's a `str`, specifically, the session ID.
```suggestion
    def get(self, session: "Optional[str]" = None) -> bytes:
```

File: otelib/pipe.py [hunk 2]
```diff
@@ -4,6 +4,8 @@
 if TYPE_CHECKING:  # pragma: no cover
     from typing import Optional
 
+    from oteapi.models import AttrDict
+
```
Comment (CasperWA) [inline otelib/pipe.py:8]:
```suggestion
```

File: otelib/backends/python/base.py [hunk 3]
```diff
@@ -133,7 +133,7 @@ def get(self, session_id: "Optional[str]" = None) -> bytes:
         """
         if session_id is None:
             session_id = f"session-{str(uuid4())}"
-            self.cache[session_id] = {}
+            self.cache[session_id] = AttrDict()
```
Comment (CasperWA) [inline otelib/backends/python/base.py:136]: It shouldn't matter whether it's an `AttrDict` or a `dict`. The purpose of the `AttrDict` is merely to wrap a standard Python `dict` in a pydantic `BaseModel`.
I.e., this change is not necessary, rendering the PR not necessary.

**Review by CasperWA — CHANGES_REQUESTED** [review]

File: otelib/pipe.py (hunk 1 above)
Comment (jesper-friis) [reply otelib/pipe.py:18]: Then the argument is very misleading. Change it to `session_id`.

Comment (jesper-friis) [reply otelib/pipe.py:18]: The SDKdemo relies on these changes. If you want to reject this PR, then please update the SDKdemo such that it runs. My calendar is fully booked from now on to December 9th.

Comment (CasperWA) [reply otelib/pipe.py:18]: Please open an issue for this outlining what changes are necessary and I'll take a look.

Comment (CasperWA) [reply otelib/pipe.py:18]: > Then the argument is misleading. Change it to `session_id`.

Agreed.

**Comment by jesper-friis** [comment]
Not needed after PR https://github.com/EMMC-ASBL/oteapi-dlite/pull/84

**Pull request closed without merging**